cyrasterize/glrasterizer.pyx

```diff
@@ -104,6 +104,32 @@ def normalize_v3(arr):
     arr[:,2] /= lens
     return arr
 
+
+def vertex_normals(points, trilist):
+    # Create a zeroed array with the same type and shape as our vertices i.e., per vertex normal
+    cdef np.ndarray[float, ndim=2, mode="c"] norm = np.zeros( (points.shape[0], points.shape[1]), dtype=np.float32)
+    tris = points[trilist]
+
+    # Calculate the normal for all the triangles, by taking the cross product of the vectors v1-v0, and v2-v0 in each triangle
+    n = np.cross( tris[::,1 ] - tris[::,0]  , tris[::,2 ] - tris[::,0] )
```

Comment:
I have a fast Cython based cross product in menpo that I was finding to be about 20% faster than numpy.